viewsb/ipc.py

```diff
@@ -7,6 +7,38 @@
 """
 
 import multiprocessing
+import time
+import threading
+import traceback
+
+
+def handle_exceptions(exception, traceback):
+    """
+    Callback that gets called when a exception is raised in a IPC process.
+    Replace with your own stub to customize behavior, for eg. in a CLI application.
+    """
+    raise exception
+
+
+class Process(multiprocessing.Process):
+    """ Process class that forwards exceptions to the parent """
+
+    def __init__(self, exception_handler, *args, **kwargs):
+        multiprocessing.Process.__init__(self, daemon=True, *args, **kwargs)
+        self._parent_conn, self._child_conn = multiprocessing.Pipe()
+        self.exception_handler = exception_handler
+
+    def run(self):
+        try:
+            multiprocessing.Process.run(self)
+        except Exception as e:
+            self.exception_handler(e, traceback.format_exc())
```

Comment:
Just to clarify, while this might seem to be running on the parent, it is actually running on the child. `Process.run` will invoke the callable passed in the constructor.

https://docs.python.org/3.8/library/multiprocessing.html#multiprocessing.Process.run